packages/ui-components/src/FormComponents/CodeBlock.tsx

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2018-2022 Elyra Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { CodeEditor } from '@jupyterlab/codeeditor';
+import { InputLabel } from '@material-ui/core';
+import { Field } from '@rjsf/core';
+import * as React from 'react';
+
+export const CodeBlock: Field = props => {
+  const label = props.schema.title ?? 'Code';
+  const codeBlockRef = React.useRef<HTMLDivElement>(null);
+  const editorRef = React.useRef<CodeEditor.IEditor>();
+
+  // `editorServices` should never change so make it a ref.
+  const servicesRef = React.useRef(props.formContext.editorServices);
+
+  React.useEffect(() => {
+    const handleChange = (args: any): void => {
+      props.onChange(args.text.split('\n'));
+    };
+
+    if (codeBlockRef.current !== null) {
+      editorRef.current = servicesRef.current.factoryService.newInlineEditor({
+        host: codeBlockRef.current,
+        model: new CodeEditor.Model({
+          value:
+            props.formData?.join('\n') ??
+            (props.schema.default as string[])?.join('\n'),
+          mimeType: servicesRef.current.mimeTypeService.getMimeTypeByLanguage({
+            name: props.formContext.language,
+            codemirror_mode: props.formContext.language
+          })
+        })
+      });
+      editorRef.current?.model.value.changed.connect(handleChange);
+    }
+
+    return (): void => {
+      editorRef.current?.model.value.changed.disconnect(handleChange);
+    };
+    // NOTE: The parent component is unstable so props change frequently causing
+    // new editors to be created unnecessarily. This effect on mount should only
+    // run on mount. Keep in mind this could have side effects, for example if
+    // the `onChange` callback actually does change.
```

Comment:
out of curiosity would it be worth adding a partial list of dependencies here for fields that we don't expect to change but we'd want to trigger an update? (ie fields not touched by the unstable parent component)